format/amf3/Tools.hx

```diff
@@ -56,9 +58,15 @@ class Tools {
 			case cast Array:
 				var o : Array<Dynamic> = o;
 				var a = new Array();
-				for(v in o)
-					a.push(encode(v));
+				for(k in Reflect.fields(o))
```

Comment:
Looking at how AMF does yes I think it requires changing AArray from AArray( a : Array<Value> ) to AArray( a : Array<Value>, ?extra : Map<String,Value> )